### DISCUSSION TITLE
Multi Action: Waiting for custom action to finish

### DISCUSSION THREAD

**Discussion by dennisrijsdijk** [elgatosf/elgato#49] [Ideas]
I'm currently working on a plugin to integrate Firebot, a Twitch chatbot. One of the actions/buttons I'm working on runs an effects list in Firebot. the function and duration of these is arbitrary. Sometimes it may be benefitial to wait for execution to finish before doing another task. During implementation and testing, I noticed it's not possible to make a custom action in Multi Action wait for completion. I know it's possible to manually add a delay action to the multi action, but since execution duration is unknown, this wouldn't work.

I've got an idea for a possible way to solve this, that would preserve backwards compatibility with current actions.

 - Add a new field in the manifest, something along the lines of `MultiActionWait` (default false)
 - Add a new event that's sent from the plugin when to continue execution, based on the context sent with a keyDown/keyUp event.

**Comment by ChekTek** [comment]
Hi @dennisrijsdijk 

Apologies this went unanswered for so long! I will take this to the Stream Deck team as a feature request. 🚀
